GetIntoTeachingApiTests/Jobs/ClaimCallbackBookingSlotJobTests.cs

```diff
@@ -0,0 +1,103 @@
+
+            _metrics.HangfireJobQueueDuration.RemoveLabelled(new[] { "ClaimCallbackBookingSlotJob" });
+            _mockContext.Setup(m => m.GetJobCreatedAt(null)).Returns(DateTime.UtcNow.AddDays(-1));
+
+            _mockAppSettings.Setup(m => m.IsCrmIntegrationPaused).Returns(false);
+        }
+
+        [Fact]
+        public void Run_OnSuccess_IncrementsNumberOfBookings()
+        {
+            var quota = new CallbackBookingQuota() { NumberOfBookings = 3, Quota = 10 };
+
+            _mockContext.Setup(m => m.GetRetryCount(null)).Returns(0);
+            _mockCrm.Setup(m => m.GetCallbackBookingQuota(_scheduledAt)).Returns(quota);
+
+            _job.Run(_scheduledAt, null);
+
+            quota.NumberOfBookings.Should().Be(4);
+
+            _mockCrm.Verify(mock => mock.Save(It.Is<CallbackBookingQuota>(q => IsMatch(quota, q))), Times.Once);
+            _mockLogger.VerifyInformationWasCalled("ClaimCallbackBookingSlotJob - Started (1/24)");
+            _mockLogger.VerifyInformationWasCalled($"ClaimCallbackBookingSlotJob - Payload {_scheduledAt}");
+            _mockLogger.VerifyInformationWasCalled($"ClaimCallbackBookingSlotJob - Succeeded - {_scheduledAt}");
+            _metrics.HangfireJobQueueDuration.WithLabels(new[] { "ClaimCallbackBookingSlotJob" }).Count.Should().Be(1);
+        }
+
+
+        [Fact]
+        public void Run_OnCallbackBookingNotAvailable_DoesNotIncrementNumberOfBookings()
+        {
+            var quota = new CallbackBookingQuota() { NumberOfBookings = 3, Quota = 3 };
+
+            _mockContext.Setup(m => m.GetRetryCount(null)).Returns(0);
+            _mockCrm.Setup(m => m.GetCallbackBookingQuota(_scheduledAt)).Returns(quota);
+
+            _job.Run(_scheduledAt, null);
+
+            _mockCrm.Verify(mock => mock.Save(It.Is<CallbackBookingQuota>(q => IsMatch(quota, q))), Times.Never);
+        }
+
+        [Fact]
+        public void Run_OnCallbackBookingNotFound_DoesNotIncrementNumberOfBookings()
+        {
+            _mockContext.Setup(m => m.GetRetryCount(null)).Returns(0);
+            _mockCrm.Setup(m => m.GetCallbackBookingQuota(_scheduledAt)).Returns<CallbackBookingQuota>(null);
+
+            _job.Run(_scheduledAt, null);
+
+            _mockCrm.Verify(mock => mock.Save(It.IsAny<CallbackBookingQuota>()), Times.Never);
+        }
+
+        [Fact]
+        public void Run_WhenCrmIntegrationPaused_Aborts()
+        {
+            _mockAppSettings.Setup(m => m.IsCrmIntegrationPaused).Returns(true);
+
+            Action action = () => _job.Run(_scheduledAt, null);
+
+            action.Should().Throw<InvalidOperationException>()
+                .WithMessage("ClaimCallbackBookingSlotJob - Aborting (CRM integration paused).");
+        }
+
+        private bool IsMatch(object objectA, object objectB)
+        {
+            objectA.Should().BeEquivalentTo(objectB);
+            return true;
```

Comment:
Can `IsMatch` return void? Or does that not work?

Reply:
It needs to evaluate to a boolean so the calling `It.Is` evaluates correctly; its wrapping a fluent assertion call as that was the easiest way to do an equivalency test that I could find (the objects need to be equal in attribute values but not the same instance of the object)